src/Metadata/Resource/Factory/ResourceMetadataFactoryTrait.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\Operation;
```

Comment:
Basically what you need to do in the
`(Annotation|Xml|Yaml)ResourceMetadataFactory` is to populate the
collectionOperations and itemOperations in the metadata object (yes,
similar to what `OperationsResourceMetadataFactory` is doing). The
`ApiLoader` will take care of the rest.
On 6 Apr 2016 23:02, "Antoine Bluchet" <notifications@github.com> wrote:

> In src/Metadata/Resource/Factory/ResourceMetadataFactoryTrait.php
> <https://github.com/api-platform/core/pull/490#discussion_r58721168>:
>
> > @@ -0,0 +1,124 @@
> > +<?php
> > +
> > +namespace ApiPlatform\Core\Metadata\Resource\Factory;
> > +
> > +use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
> > +use ApiPlatform\Core\Metadata\Resource\Operation;
>
> Yes, but by defining custom operations as arrays (through configurations
> or annotations), you've already defined everything needed to build a Route.
> Therefore, you won't go through OperationResourceMetadataFactory but you
> won't need any Operation or Pagination.
>
> For example this entity
> <https://github.com/api-platform/core/blob/master/tests/Fixtures/TestBundle/Entity/RelationEmbedder.php#L30>
> works fine, and doesn't go through the highlighted code path because it
> doesn't need to.
>
> Check also ApiLoader
> <https://github.com/api-platform/core/blob/master/src/Bridge/Symfony/Routing/ApiLoader.php#L92>
> where $operation is a simple Array with no pagination nor based on any
> Operation interface.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/api-platform/core/pull/490/files/37f58321cc8415e7f464031ee8431b73aa20417d#r58721168>
>